superset-frontend/src/SqlLab/components/EstimateQueryCostButton/index.tsx

```diff
@@ -93,7 +91,7 @@ const EstimateQueryCostButton = props => {
         triggerNode={
           <Button
             style={{ height: 32, padding: '4px 15px' }}
-            onClick={onClick}
+            onClick={getEstimate()}
```

Comment:
onClick expects a function (not the return value of a function) so you need to remove the parens here

```suggestion
            onClick={getEstimate}
```

Reply:
That's right. Thanks for the suggestion.

Reply:
@etr2460 Looks like this is going to fail. I checked the type declaration for `onClick` prop locally in my text editor (i.e. VS Code) and it seems like `onClick` expects an event handler method that is passed as a call back function: 

Here is the error I get after changing the prop to `onClick={getEstimate}`: 

```
Type 'Function' is not assignable to type 'OnClickHandler'.
  Type 'Function' provides no match for the signature '(event: MouseEvent<HTMLElement, MouseEvent>): void'.ts(2322)
index.tsx(47, 3): The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & ButtonProps'
(JSX attribute) ButtonProps.onClick?: OnClickHandler | undefined

```

So, I guess we should restore that other `onClick` method that we removed thinking that it was just there to call the 'getEstimate' method, and may be we can rename it to `onClickHandler` to avoid any confusion in future.

Reply:
yeah, i guess that makes sense :'( maybe add a comment on why the handler is needed